wellplated/models.py

```diff
@@ -0,0 +1,291 @@
+    'well-right-column-gte-1': GreaterThanOrEqual(
+        Cast(Right('label', 2), PositiveSmallIntegerField()), 1
+    ),
+    'well-right-column-lte-format-max': LessThanOrEqual(
+        Right('label', 2), Cast(Substr('container', 2, 4), PositiveSmallIntegerField())
+    ),
+    'well-container-code-label-length-19': Q(
+        # dot, label row, label column
+        container_code_label__length=CONTAINER_CODE_LENGTH + 1 + 1 + 2
+    ),
+    'well-container-code-label-has-1-dot': Q(
+        container_code_label__length=(
+            Length(Replace('container_code_label', Value('.'), Value(''))) + 1
+        )
+    ),
+}
+
+
+class Well(Model):
+    """
+    One of multiple positions on a plate, or the singular position of a tube.
+
+    Positions are stored and displayed as "Battleship notation" labels with alphabetical row and
+    numerical column, the latter zero-padded for easy sorting and consistent length.
+
+    1-well   tube/trough  coordinate is   A01
+    6-well   plate coordinates range from A01 to B03
+    24-well  plate coordinates range from A01 to D06
+    96-well  plate coordinates range from A01 to H12
+    384-well plate coordinates range from A01 to P24
+    """
+
+    container = ParentalKey(
+        Container,
+        db_column='container_code',
+        editable=False,
+        on_delete=PROTECT,
+        related_name='wells',
+        to_field='code',
+    )
+    label = CharField(editable=False, max_length=3)
+    container_code_label = GeneratedField(
+        db_persist=True,
+        editable=False,
+        expression=Concat('container', Value('.'), 'label'),
+        # dot, label row, label column
+        output_field=CharField(max_length=CONTAINER_CODE_LENGTH + 1 + 1 + 2),
+        unique=True,
+    )
+
+    sinks: ManyToManyField[Self, Self] = ManyToManyField(
+        'self',
+        through='Transfer',
+        through_fields=('source', 'sink'),
+        symmetrical=False,
+        related_name='sources',
+    )
+
+    objects = WellManager()
+
+    class Meta(TypedModelMeta):
+        constraints = [  # noqa: RUF012
+            CheckConstraint(condition=condition, name=name)
+            for name, condition in well_checks.items()
+        ]
+
+    def __str__(self) -> str:
+        return f'{self.container_code_label}'
+
+
+@register_snippet
+class Plan(Model):
+    """
+    A set of transfers describing what should happen.
+
+    See results for measurements or assumptions of what actually happened.
+
+    Plans with no results are the todo list.
+
+    Plans with one result are the common finished case.
+
+    Plans may have multiple results when reworking with the same containers.
+
+    Create a new plan to rework using different containers.
+    """
+
+    created_at = DateTimeField(auto_now_add=True)
+    created_by = ForeignKey(User, on_delete=PROTECT, related_name='plans_created')
+
+    def __str__(self) -> str:
+        return f'plan {self.pk}'
+
+
+@register_snippet
+class Transfer(Model):
```

Comment:
Maybe name this `WellTransfer` since it's specific to wells, but this is up to you

Reply:
I'll think about that